src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/BaseDiagnosticItemSource.cs

```diff
@@ -110,7 +109,9 @@ private IEnumerable<BaseDiagnosticItem> GetDiagnosticItems(string language, Comp
 
             Contract.ThrowIfFalse(HasItems);
```

Comment:
This was the problematic assert: conceptually the idea was "we shouldn't be trying to create items if we know we don't have any to show", but in the code prior to this new change we would assign the field before calling GetDiagnosticItems, so HasItems would instead return false because we had realized the list prematurely.